streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -661,6 +673,131 @@ public void statefulTopologyShouldCreateStateDirectory() throws Exception {
         startStreamsAndCheckDirExists(topology, asList(inputTopic, globalTopicName), outputTopic, true);
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldInitializedUserSerdes() {
+        final Deserializer mockSourceKeyDeserialzer = mock(Deserializer.class);
+        mockSourceKeyDeserialzer.configure(anyObject(), eq(true));
+        expectLastCall();
+        final Deserializer mockSourceValueDeserialzer = mock(Deserializer.class);
+        mockSourceValueDeserialzer.configure(anyObject(), eq(false));
+        expectLastCall();
+
+        final Serde mockSourceKeySerde = mock(Serde.class);
+        final Serde mockSourceValueSerde = mock(Serde.class);
+        expect(mockSourceKeySerde.deserializer()).andReturn(mockSourceKeyDeserialzer).anyTimes();
+        expect(mockSourceValueSerde.deserializer()).andReturn(mockSourceValueDeserialzer).anyTimes();
+
+
+
+        final Serializer mockThroughKeySerializer = mock(Serializer.class);
+        mockThroughKeySerializer.configure(anyObject(), eq(true));
+        expectLastCall();
+        final Serializer mockThroughValueSerializer = mock(Serializer.class);
+        mockThroughValueSerializer.configure(anyObject(), eq(false));
+        expectLastCall();
+        final Deserializer mockThroughKeyDeserializer = mock(Deserializer.class);
+        mockThroughKeyDeserializer.configure(anyObject(), eq(true));
+        expectLastCall();
+        final Deserializer mockThroughValueDeserializer = mock(Deserializer.class);
+        mockThroughValueDeserializer.configure(anyObject(), eq(false));
+        expectLastCall();
+
+        final Serde mockThroughKeySerde = mock(Serde.class);
+        final Serde mockThroughValueSerde = mock(Serde.class);
+        expect(mockThroughKeySerde.serializer()).andReturn(mockThroughKeySerializer).anyTimes();
+        expect(mockThroughValueSerde.serializer()).andReturn(mockThroughValueSerializer).anyTimes();
+        expect(mockThroughKeySerde.deserializer()).andReturn(mockThroughKeyDeserializer).anyTimes();
+        expect(mockThroughValueSerde.deserializer()).andReturn(mockThroughValueDeserializer).anyTimes();
```

Comment:
should this expect be aligned with the above 'expect's ?

Reply:
What do you mean exactly? What lines are you referring to?

Reply:
In the current PR, these are aligned.
Dropping former comment.